### PR TITLE
examples/docs: added spring project

### PR DESCRIPTION
**Pull request by dorav**
Adds a Spring(https://spring.io/) example project

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -39,3 +39,14 @@ jobs:
...
+      - uses: actions/setup-java@v3
+        with:
+          distribution: zulu
+          java-version: 17
+
+      - name: Integration test spring project
+        run: |
+          cd examples/with_spring_gradle
+          ./gradlew -q bootRun | tee schema_output
+          diff schema_output src/test/expected_default_schema
```
Comment (rotemtam) [inline .github/workflows/ci.yaml:52]: Assertions should be on Atlas being able to consume and plan migrations, a better way here is to use atlas migrate diff and make sure no change is necessary.

Comment (dorav) [reply .github/workflows/ci.yaml:52]: Let's do it in a different PR? This is not the only test that will need to change

Comment (rotemtam) [reply .github/workflows/ci.yaml:52]: I prefer to merge the tests with the feature. You can fix the rest in a different PR

Comment (dorav) [reply .github/workflows/ci.yaml]: completed this, resolving

**Review by rotemtam — APPROVED** [review]

**Pull request merged**